fil_fungible_token/src/blockstore.rs

```diff
@@ -42,31 +44,51 @@ impl fvm_ipld_blockstore::Blockstore for Blockstore {
     }
 }
 
-// TODO: put this somewhere more appropriate when a tests folder exists
-/// An in-memory blockstore impl taken from filecoin-project/ref-fvm
+/// An in-memory blockstore impl that shares underlying memory when cloned
```

Comment:
I _think_ it is ok to assume this is how a blockstore implementation should work. a blockstore impl running on-chain should persist everything to the ipld layer directly making the actual blockstore handle "stateless" from the pov of a clone